pkg/apis/resources/v1alpha1/types.go

```diff
@@ -169,6 +169,9 @@ type ManagedResourceStatus struct {
 	// Resources is a list of objects that have been created.
 	// +optional
 	Resources []ObjectReference `json:"resources,omitempty"`
+	// SecretsChecksum is the checksum of referenced secrets data.
+	// +optional
+	SecretsChecksum string `json:"secretsChecksum,omitempty"`
```

Comment:
```suggestion
	SecretsChecksum *string `json:"secretsChecksum,omitempty"`
```

Nit: Should we call it `SecretsDataChecksum` to make clear it's really only about `data`?